source/agora/consensus/EnrollmentManager.d

```diff
@@ -587,4 +672,13 @@ unittest
     man.getUnregisteredEnrollments(enrolls);
     assert(enrolls.length == 3);
     assert(enrolls.isStrictlyMonotonic!("a.utxo_key < b.utxo_key"));
+
+    // test serialization/deserializetion for pre-images
+    auto preimages_1 = man.getPreimages();
+    auto preimages_2 = man.loadPreimages();
+    int true_count;
+    foreach (i; 0 .. 1008)
```

Comment:
These few lines can be replaced with `assert(preimages_1[] == preimages_2[]);`. This will check the length too.